### PR TITLE
Remove plugins and speed up compilation of AnalysisController

### PR DESCRIPTION
**Pull request by MMory**

### CODE REVIEW

File: include/phasar/Controller/AnalysisController.h [hunk 1]
```diff
@@ -81,6 +87,67 @@ class AnalysisController {
...
+  template <typename AnalysisTy, bool WithConfig = false>
+  void executeIDEAnalysis() {
+    executeAnalysis<IDESolver_P<AnalysisTy>, AnalysisTy, WithConfig>();
+  }
+
+  template <class Solver_P, typename AnalysisTy, bool WithConfig>
+  void executeAnalysis() {
+    if constexpr (WithConfig) {
+      std::string AnalysisConfigPath =
+          (0 < AnalysisConfigs.size()) ? AnalysisConfigs[0] : "";
```
Comment (fabianbs96) [inline include/phasar/Controller/AnalysisController.h:132]:
```suggestion
          !AnalysisConfigs.empty() ? AnalysisConfigs[0] : "";
```

File: include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/EdgeFact.h [hunk 2]
```diff
@@ -17,7 +17,6 @@ class raw_ostream;
 namespace psr {
 
 /// A common superclass of edge-facts used by non-template IDETabulationProblems
-/// (for example in plugins)
```
Comment (fabianbs96) [inline include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/EdgeFact.h:20]: This file can probably be removed

File: include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/FlowFact.h [hunk 3]
```diff
@@ -16,7 +16,7 @@
 namespace psr {
 
 /// A common superclass of dataflow-facts used by non-template tabulation
-/// problems (for example in plugins)
```
Comment (fabianbs96) [inline include/phasar/PhasarLLVM/DataFlowSolver/IfdsIde/FlowFact.h:19]: This file can probably be removed

**Review by fabianbs96 — COMMENTED** [review]
Looks already very good!

I have found a few things that can be removed:
- ${PHASAR_PLUGINS_LIB} from tools/example-tool/CMakeLists.txt and tools/boomerang/CMakeLists.txt
- .travis.yml:87
- .gitlab-ci.yml:26-27

**Review by vulder — APPROVED** [review]

**Pull request merged**